build.gradle

```diff
@@ -271,8 +278,10 @@ subprojects {
             '-Djavax.xml.stream.XMLInputFactory=com.sun.xml.internal.stream.XMLInputFactoryImpl',
             "-Dversion.id=${project.version}",
             '-ea',
-            '-Djunit.jupiter.execution.parallel.config.strategy=fixed',
-            '-Djunit.jupiter.execution.parallel.config.fixed.parallelism=2'
+            '-Djunit.jupiter.execution.parallel.enabled=true',
+            '-Djunit.jupiter.execution.parallel..mode.default=concurrent'
```

Comment:
double dot seems incorrect

Reply:
> So we should avoid running into the same problems again. Can we have at least a way to configure the run in CI to not run with full parallel build so we can keep CI stable without adding such hard to track-down breakages?

Thank you for the feedback. It is definitely possible, although I have one major question: How to detect the "CI" and to know that we are building on the "CI"? 
I have built everything just fine on a 2 Core 3GB machine, is the "CI" anyhow worse equipped? Does the "CI" have a certain distinct hostname or configuration we could use for identifying?

Reply:
Would it be possible to rely on an Environment Variable, e. g.:
```
boolean isJenkinsBuild = Boolean.valueOf(System.getenv("CI_BUILD"));
boolean isSoftwarePresent = new ProcessBuilder("check software presence").start().waitFor() == 0;
Assume.assumeTrue("Software not present", isSoftwarePresent || isJenkinsBuild);
```
for restricting the build to serial mode?

Would you be able and willing to set the Environment Variable "CI_BUILD" when kicking off Jenkins:
```
pipeline {
            stage('test'){ 

                    sh '''CI_BUILD="TRUE" // setting the env variable in the same shell where you are running mvn
                        mvn test'''
             }
        }
```

Any better idea?

Reply:
The CI build uses the 'jenkins' task in build.gradle while human users would probably use 'test' task. So, we can have the build.gradle behave differently depending on the task that is called.

Reply:
I'd rather keep the "jenkins" task useable and quick locally for developers as well so we can use it as a full local "pre-commit-check".

Reply:
I do not think, I damaged this `jenkins` task, I only let it set the global `CI_BUILD` variable (in case it has not been set on the command line).

Reply:
Sorry, wrong word "useable", I rather meant to keep the env-variable as single place which decides how to run the build. 

If I run "jenkins" locally it would be good to still be able to decide if I want to get the speedup of the parallel building.